packages/twenty-front/src/modules/object-record/record-table/hooks/useRecordTableMoveFocus.ts

```diff
@@ -17,58 +31,67 @@ export const useRecordTableMoveFocus = (recordTableId?: string) => {
     recordTableId,
   );
 
+  const tableAllRowIdsState = useRecoilComponentCallbackStateV2(
+    tableAllRowIdsComponentState,
+    recordTableId,
+  );
+
   const moveUp = useRecoilCallback(
     ({ snapshot }) =>
       () => {
+        const allRowIds = getSnapshotValue(snapshot, tableAllRowIdsState);
         const softFocusPosition = getSnapshotValue(
           snapshot,
           softFocusPositionState,
         );
 
-        let newRowNumber = softFocusPosition.row - 1;
+        const currentRowIndex = allRowIds.indexOf(softFocusPosition.recordId);
 
-        if (newRowNumber < 0) {
-          newRowNumber = 0;
+        if (currentRowIndex === -1) {
+          return;
+        }
+
+        let newRowIndex = currentRowIndex - 1;
+
+        if (newRowIndex < 0) {
+          newRowIndex = 0;
         }
 
         setSoftFocusPosition({
           ...softFocusPosition,
-          row: newRowNumber,
+          recordId: getRecordIdFromRowIndex(allRowIds, newRowIndex),
```

Comment:
ok, so the soft focus is based on the recordId now and not on the rowIndex anymore? I feel we could keep using the rowIndex as we still all the allRowIds state